### PR TITLE
Merge flutter frontend to modelgen staging branch

### PR DESCRIPTION
**Pull request by AaronZyLee**
*Issue #, if available:*

*Description of changes:*
Merge the flutter frontend plugin to modelgen staging branch so that developers can add flutter type frontend and run test on modelgen.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-core/src/init/initProjectHelper.ts [hunk 1]
```diff
@@ -133,6 +133,7 @@ export function initIosProjectWithProfile(cwd: string, settings: Object) {
       .wait("Choose the type of app that you're building")
       .send('j')
```
Comment (attilah) [inline packages/amplify-e2e-core/src/init/initProjectHelper.ts:134]: We added the `keyDown` and other methods with a `count` parameter, please update all the tests to use those and no `'j'`.

File: packages/amplify-e2e-core/src/init/initProjectHelper.ts [hunk 2]
```diff
@@ -149,6 +150,41 @@ export function initIosProjectWithProfile(cwd: string, settings: Object) {
...
+      .sendLine(s.editor)
+      .wait("Choose the type of app that you're building")
+      .send('j')
+      .send('j')
+      .sendCarriageReturn()
+      .wait('Where do you want to store your configuration file')
+      .sendLine('./lib/')
+      .wait('Using default provider  awscloudformation')
+      .wait('Do you want to use an AWS profile?')
+      .sendLine('y')
```
Comment (attilah) [inline packages/amplify-e2e-core/src/init/initProjectHelper.ts:172]: We added the `confirmYes` and `confirmNo` please use those.

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts [hunk 3]
```diff
@@ -17,14 +17,23 @@ describe('amplify add analytics', () => {
...
     const rightName = 'myapp';
     await addPinpoint(projRoot, { rightName, wrongName: '$' });
     await amplifyPushUpdate(projRoot);
     expect(fs.existsSync(path.join(projRoot, 'amplify', 'backend', 'analytics', rightName))).toBe(true);
   });
 
+  it('add pinpoint for flutter', async () => {
+    await initFlutterProjectWithProfile(projRoot, { name: 'storageTest' });
+    const rightName = 'myapp';
+    await addPinpoint(projRoot, { rightName, wrongName: '$' });
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/__tests__/analytics.test.ts:31]: No need to test `wrongName` here. Please remove.

File: packages/amplify-e2e-tests/src/__tests__/auth_1.test.ts [hunk 4]
```diff
@@ -61,4 +71,103 @@ describe('amplify add auth...', () => {
     await addAuthWithDefault(projRoot, {});
     await amplifyPushAuth(projRoot);
   });
+
+  it('...should init a project and add auth with defaultSocial', async () => {
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/__tests__/auth_1.test.ts:75]: What is the difference in these tests compared to the ones in `auth_2.test.ts` ?

File: packages/amplify-frontend-flutter/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Change Log
```
Comment (attilah) [inline packages/amplify-frontend-flutter/CHANGELOG.md:1]: How is this new? Did we rename the package or coming from a wrong merge/rebase? PRs should not have changelog files.

File: packages/amplify-frontend-flutter/commands/flutter.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (attilah) [inline packages/amplify-frontend-flutter/commands/flutter.js:1]: What is this?

File: packages/amplify-frontend-flutter/package.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  "author": "Amazon Web Services",
+  "license": "Apache-2.0",
+  "main": "index.js",
+  "keywords": [
+    "graphql",
+    "appsync",
+    "aws"
+  ],
+  "dependencies": {
+    "chalk": "^3.0.0",
```
Comment (attilah) [inline packages/amplify-frontend-flutter/package.json:19]: make sure that these dependency versions are aligned with other packages, like `inquirer` is `7.3.3` currently.

File: packages/amplify-frontend-flutter/lib/configuration-manager.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    Object.assign(context.exeInfo.projectConfig[constants.Label], inputParams);
+  } else if (!context.exeInfo.inputParams.yes) {
+    context.print.info('Please tell us about your project');
+    const { config } = context.exeInfo.projectConfig[constants.Label];
+
+    context.print.warning('⚠️  Flutter project support in the Amplify CLI is in DEVELOPER PREVIEW.');
+    context.print.warning('Only the following categories are supported:');
+    context.print.warning(' * Auth');
+    context.print.warning(' * Analytics');
+    context.print.warning(' * Storage');
```
Comment (attilah) [inline packages/amplify-frontend-flutter/lib/configuration-manager.js:52]: In Analytics, we have Kinesis there, which is not supported, only Pinpoint, right? Perhaps we should pointing it out.

Also why do we have API, Predictions related code (`constructPredictions`) if that is not supported? Please remove the excess.

**Review by attilah — CHANGES_REQUESTED** [review]
Overall looks good as it is 90% identical to the frontend category plugins.

A few changes I like to see:
- Put all the Flutter related E2E tests into 1 test file within the E2E project to make separation easier later and don't spread across categories.
- Reorg the plugin source code into `src` folder with `lib` output and add TypeScript passthrough compilation, like we have in other some packages like `amplify-provider-awscloudformation` as an example.

**Comment by AaronZyLee** [comment]
Close this PR since #5849 is merging flutter frontend to master.

File: packages/amplify-frontend-flutter/commands/flutter.js (hunk 6 above)
Comment (haverchuck) [reply packages/amplify-frontend-flutter/commands/flutter.js:1]: This is a simple command to see whether the plugin exists with 'amplify flutter' in the terminal. It is present for all other frontend plugins. I'm not sure that it is really necessary, but is included for consistency.

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts (hunk 3 above)
Comment (haverchuck) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts:31]: The test actually fails if I remove this.

File: packages/amplify-e2e-core/src/init/initProjectHelper.ts (hunk 2 above)
Comment (haverchuck) [reply packages/amplify-e2e-core/src/init/initProjectHelper.ts:172]: Done.

File: packages/amplify-e2e-core/src/init/initProjectHelper.ts (hunk 1 above)
Comment (haverchuck) [reply packages/amplify-e2e-core/src/init/initProjectHelper.ts:134]: Done.

File: packages/amplify-frontend-flutter/CHANGELOG.md (hunk 5 above)
Comment (haverchuck) [reply packages/amplify-frontend-flutter/CHANGELOG.md:1]: Not an issue in PR.

File: packages/amplify-frontend-flutter/package.json (hunk 7 above)
Comment (haverchuck) [reply packages/amplify-frontend-flutter/package.json:19]: Inquirer dependency verision is fixed; all others verified.

File: packages/amplify-frontend-flutter/lib/configuration-manager.js (hunk 8 above)
Comment (haverchuck) [reply packages/amplify-frontend-flutter/lib/configuration-manager.js:52]: Done.

**Pull request closed without merging**